evm_loader/program/src/evm/buffer.rs

```diff
@@ -221,17 +209,116 @@ impl<'de> serde::Deserialize<'de> for Buffer {
 
                 let (index, variant) = data.variant::<u32>()?;
                 match index {
-                    0 => variant.unit_variant().map(|_| Buffer::empty()),
                     1 => variant.newtype_variant().map(Buffer::from_slice),
                     2 => variant.struct_variant(&["key", "range"], self),
                     _ => Err(serde::de::Error::unknown_variant(
                         "_",
-                        &["empty", "owned", "account"],
+                        &["owned", "account"],
                     )),
                 }
             }
         }
 
-        deserializer.deserialize_enum("evm_buffer", &["empty", "owned", "account"], BufferVisitor)
+        deserializer.deserialize_enum("evm_buffer", &["owned", "account"], BufferVisitor)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::cell::RefCell;
+    use std::rc::Rc;
+
+    macro_rules! assert_slice_ptr_eq {
+        ($actual:expr, $expected:expr) => {{
+            let actual: &[_] = $actual;
+            let (expected_ptr, expected_len): (*const _, usize) = $expected;
+            assert_eq!(actual.as_ptr(), expected_ptr);
+            assert_eq!(actual.len(), expected_len);
+        }};
+    }
```

Comment:
I believe macros should be used only as a last resort when generics wouldn't do the job. This macro can be replaced by a generic method:
```Rust
    fn assert_slice_ptr_eq<T>(actual: &[T], (expected_ptr, expected_len): (*const T, usize)) {
        assert_eq!(actual.as_ptr(), expected_ptr);
        assert_eq!(actual.len(), expected_len);
    }
```

Reply:
I chose a macro so the line number is more useful when the assertions fail.